src/extension/infer.rs

```diff
@@ -516,10 +516,12 @@ impl UnificationContext {
                                 solved = true;
                             }
                         };
-                    } else if let Some(superset) = self.get_solution(&meta) {
-                        let subset = ExtensionSet::singleton(r).missing_from(superset);
-                        self.add_solution(self.resolve(*other_meta), subset);
-                        solved = true;
+                    } else if let Some(_superset) = self.get_solution(&meta) {
+                        // Here, we're stuck because the Plus constraint only
+                        // signifies a preorder. I.e. if m0 = m1 + 'R', it's
```

Comment:
It might be nicer still to use `meta` and `other_meta` instead of `m0` and `m1` (whichever way around it is)